### PR TITLE
Component router guide

### PR DESCRIPTION
**Pull request by petebacondarwin**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

docs update!
- **Other information**:

So this is still a draft but it could be good enough to merge and iterate on.
The big problem is that, at the moment, I had to include the whole of the router code in the runnable example.

You can see a demo of the guide here: https://ci.angularjs.org/job/angular.js-pete/1073/artifact/build/docs/guide/component-router

### CODE REVIEW

**Comment by gkalpak** [comment]
Looks good overall (there are a few typis here and there, but the diff is not displayed to comment :disappointed:)

It might be better to remove the Crisis Center bits from the example, since it's not mentioned in the guide. (Maybe provide the whole app (**with** Crisis Center) at the end as a more complete example.

--
Regarding including the whole router code in the example, it's ugly, but I can live with it as a temporary work-around.
When the issue with `v0.2.x` being marked as pre-release (instead of latest) is resolved, we could use https://npmcdn.org maybe

File: src/ngComponentRouter/Router.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+/**
+ * @ngdoc directive
+ * @name  ngOutlet
+ * @description
+ *
+ */
+
+/**
+ * @name ngLink
```
Comment (brandonroberts) [inline src/ngComponentRouter/Router.js]: The sample code used here for configuring routes and the ngLink is incorrect.

Comment (petebacondarwin) [inline src/ngComponentRouter/Router.js]: Yes - work in progress :-)

**Comment by btford** [comment]
FYI I published `@angular/router` v0.2.0 from master– you should be able to use that now.

**Comment by petebacondarwin** [comment]
Yes!

**Comment by petebacondarwin** [comment]
npmcdn doesn't seem to cope with the scoped package `@angular/router`

**Comment by gkalpak** [comment]
I can see the files just fine (e.g. [angular_1_router.js][1]).
[1]: https://npmcdn.com/@angular/router@0.2.0/angular1/angular_1_router.js

**Pull request merged**